### PR TITLE
Fix for not found lesson check

### LINKED ISSUE #428: Wrong handling not found lesson for assigning mentor
When we try to assign mentor for non-existent lesson we get an NullReferenceException. So we need to add validation for that case
Also we need to change check for null mentorId in LessonService (AssignMentorToLessonAsync)

### PR DESCRIPTION
**Pull request by Serhii-Dudar**
Validation for not found lesson were added
Changed check for null mentorId

### CODE REVIEW

**Comment by gingerbread-monster** [comment]
Please, always provide "steps to reproduce" information for issues/PRs marked as bugs. Adding a screenshot would probably be a good practice too.

File: CharlieBackend.Business/Services/LessonService.cs [hunk 1]
```diff
@@ -145,10 +145,15 @@ public async Task<Result<Lesson>> AssignMentorToLessonAsync(AssignMentorToLesson
...
             if (mentorToAssign == null)
             {
-                throw new NullReferenceException();
+                return Result<Lesson>.GetError(ErrorCode.NotFound, $"Mentor with id {ids.MentorId} is not found");
             }
             var foundLesson = await _unitOfWork.LessonRepository.GetByIdAsync(ids.LessonId);
 
+            if (foundLesson == null)
+            {
+                return Result<Lesson>.GetError(ErrorCode.NotFound, $"Lesson with id {ids.LessonId} is not found");
```
Comment (gingerbread-monster) [inline CharlieBackend.Business/Services/LessonService.cs:154]: Please also update a corresponding controller method's summary to match this updated result.

**Comment by gingerbread-monster** [comment]
Also these changes would probably affect this pull request: https://github.com/ita-social-projects/WhatBackend/pull/408

**Review by andy-dendy-repo — APPROVED** [review]

**Review by TetianaDomanska-dev — APPROVED** [review]

File: CharlieBackend.Business/Services/LessonService.cs (hunk 1 above)
Comment (Serhii-Dudar) [reply CharlieBackend.Business/Services/LessonService.cs:154]: Update here https://github.com/ita-social-projects/WhatBackend/pull/429/commits/71949395e9212d1f529055cdc999982831a28a87

**Review by Serwlashev — APPROVED** [review]

**Pull request merged**